pkg/sdk/connections_def.go

```diff
@@ -0,0 +1,121 @@
+package sdk
+
+//go:generate go run ./poc/main.go
+
+import (
+	g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
+)
+
+var ConnectionDef = g.NewInterface(
+	"Connections",
+	"Connection",
+	g.KindOfT[AccountObjectIdentifier](),
+).CreateOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/create-connection",
+	g.NewQueryStruct("CreateConnection").
+		Create().
+		SQL("CONNECTION").
+		IfNotExists().
+		Name().
+		OptionalComment().
+		WithValidation(g.ValidIdentifier, "name"),
+).CustomOperation(
+	"CreateReplicated",
+	"https://docs.snowflake.com/en/sql-reference/sql/create-connection",
+	g.NewQueryStruct("CreateReplicated").
+		Create().
+		SQL("CONNECTION").
+		IfNotExists().
+		Name().
+		Identifier("ReplicaOf", g.KindOfT[ExternalObjectIdentifier](), g.IdentifierOptions().Required().SQL("AS REPLICA OF")).
+		OptionalComment().
+		WithValidation(g.ValidIdentifier, "name").
+		WithValidation(g.ValidIdentifier, "ReplicaOf"),
+).CustomOperation(
```

Comment:
nit: IMO in this case having these custom operations is a bit of an overkill (each has just one option, so they could be all hidden behind a single Alter with good exclusion validation)

Reply:
You're right, it will be easier to have just the basic operation with exclusion validations